Add metadata tag and acknowledgement to PTCs

mrmundt opened the pull request:
As part of the PSIP automation process, we want to add some metadata that can be traced (via github searches) when a project uses one of the catalog cards.

This PR recommends adding both metadata to the top (which will not render on GitHub or GitLab but will be in the raw source code) and an acknowledgement at the bottom. In this way, if a project is copy-pasting an existing PTC into their own repo, it should be searchable in two ways.

**NOTE**: The choice `ecp-psip-ptc` is for uniqueness. In searching all of GitHub, I didn't find a use of that tag or phrase anywhere that is unrelated to ECP.

EDIT: Partially resolves #41

elaineraybourn commented:
@mrmundt thanks for your progress on this. One thing to consider -- We are moving along the path of PSIP automation so the toolkit outlives the ECP project. So...as you have "ecp" in the metadata tag -- how would you like to address this? Thanks!

mrmundt commented:
@elaineraybourn - That's a good question. I originally had the tag as `psip-ptc`, but that is not unique across all of GitHub. That is, there are other repositories and projects that use that acronym (not a lot, but they are there!). Is there an inherent problem in saying, "This was originally created through efforts by ECP"? If there isn't, I don't see an outright problem with `ecp-psip-ptc`.

That being said, another option I like is `bssw-psip-ptc` because a lot of the info is hosted in the `bssw-psip` organization and GitHub Page. Thoughts?

elaineraybourn commented:
I like bssw-psip-ptc better. Bssw won't change soon.

mrmundt commented:
@elaineraybourn - Changed!

Any thoughts on the acknowledgement language?

elaineraybourn commented:
@mrmundt can you make a few suggestions to choose from? What would Uncle Bob do? WWUBD. :-)

mrmundt commented:
@elaineraybourn - Sorry, I forget that you're not primarily a software developer! If you click on "Files changed", you can see that I added some language already based on what I thought was good. The language I added to each file was:

### Acknowledgement

This Project Tracking Card originated from the [PSIP PTC Catalog](https://bssw-psip.github.io/ptc-catalog/). The development of the PSIP PTC Catalog was supported by the Exascale Computing Project (17-SC-20-SC), a collaborative effort of the U.S. Department of Energy Office of Science and the National Nuclear Security Administration.
